jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/Content.java

```diff
@@ -155,6 +155,24 @@ public static void copy(Source source, Sink sink, Chunk.Processor chunkProcessor
      */
     public interface Source
     {
+        /**
+         * Factory of {@link Content.Source}.
+         */
+        interface Factory
+        {
+            /**
+             * Creates a new {@link Content.Source}.
+             *
+             * @param bufferPool the {@link ByteBufferPool} to get buffers from. null means allocate new buffers as needed.
+             * @param bufferSize the size of the buffer to be used for the copy. Any value &lt; 1 means use a default value.
+             * @param direct the directness of the buffers, this parameter is ignored if {@code bufferSize} is &lt; 1.
```

Comment:
why `bufferSize > 1`? 

I suggest rephrase it as `The desired "directness" of the buffers from the {@code bufferPool}"

Reply:
I've changed the signature to take a `ByteBufferPool.Sized` instead which should make things clearer.